cfme/containers/container.py

```diff
@@ -36,6 +36,12 @@ def __init__(self, name, pod):
         self.name = name
         self.pod = pod
 
+    def _on_detail_page(self):
+        """ Returns ``True`` if on the providers detail page, ``False`` if not."""
+        ensure_browser_open()
```

Comment:
This shouldn't ever be needed in this kind of model. ```force_navigate``` is the preferred method of "navigating" to a page. If your intention is to avoid navigating if you don't have to, the best way would be to use a try/except clause on the ```is_displayed``` call and return False. Your subsequent calls can then ```force_navigate``` which will open the browser if it is closed.